src/components/NodeList.js

```diff
@@ -18,55 +19,100 @@ const EnhancedNode = DragSource(selectable(Node));
-
-  const classNames = cx(
-    'node-list',
-    { 'node-list--hover': !isSource && willAccept && isOver },
-    { 'node-list--drag': !isSource && willAccept }, // TODO: rename class
-  );
-
-  return (
-    <StaggeredTransitionGroup
-      className={classNames}
-      component="div"
-      delay={animation.duration.fast * 0.2}
-      duration={animation.duration.slow}
-      start={animation.duration.slow + 3}
-      transitionName="node-list--transition"
-      transitionLeave={false}
-    >
-      { isOver && willAccept &&
-        <Node key="placeholder" placeholder />
-      }
-      {
-        nodes.map(node => (
-          <span key={node.uid}>
-            <EnhancedNode
-              color={nodeColor}
-              label={label(node)}
-              selected={selected(node)}
-              onSelected={() => onSelect(node)}
-              meta={() => ({ ...node, itemType })}
-              {...node}
-            />
-          </span>
-        ))
-      }
-    </StaggeredTransitionGroup>
-  );
-};
+class NodeList extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      nodes: props.nodes,
+      refresh: 0,
+      stagger: true,
+    };
+
+    this.refreshTimer = null;
+  }
+
+  componentWillReceiveProps(newProps) {
+    // Don't update if nodes are the same
+    if (isEqual(map(newProps.nodes, 'uid'), map(this.state.nodes, 'uid'))) {
+      return;
+    }
+
+    // if we are on the same prompt/provided same id, then just append the node
+    if (newProps.id === this.props.id) {
+      this.setState({ nodes: newProps.nodes, stagger: false });
+      return;
+    }
+
+    // Otherwise, transition out and in again
+    this.props.scrollTop(0);
+
+    this.setState(
+      { nodes: [] },
+      () => {
+        if (this.refreshTimer) { clearTimeout(this.refreshTimer); }
+        this.refreshTimer = setTimeout(
+          () => this.setState({
+            nodes: newProps.nodes,
+            stagger: true,
+          }),
+          animation.duration.slow,
+        );
+      },
+    );
+  }
+
+  render() {
+    const {
+      nodeColor,
+      label,
+      selected,
+      onSelect,
+      itemType,
+      isOver,
+      willAccept,
+      meta,
+    } = this.props;
+
+    const {
+      stagger,
+      nodes,
+    } = this.state;
+
+    const isSource = !!find(nodes, ['uid', get(meta, 'uid', null)]);
+
+    const classNames = cx(
+      'node-list',
+      { 'node-list--hover': !isSource && willAccept && isOver },
+      { 'node-list--drag': !isSource && willAccept }, // TODO: rename class
+    );
+
+    return (
+      <TransitionGroup
+        className={classNames}
+      >
+        {
+          nodes.map((node, index) => (
+            <NodeTransition
+              key={`${node.uid}`}
+              index={index}
+              stagger={stagger}
+            >
+              <EnhancedNode
+                color={nodeColor}
+                label={`${label(node)}_${index}`}
```

Comment:
The names with the _[number] is certainly disconcerting when the numbers bounces around based on which panel the node is in!